## Likelihood Ratio

# Intuition


- **Likelihood Ratio (LR)**: The ratio of the likelihood for two models

**FIGURE PLACEHOLDER:** ![Likelihood Intuition Cartoon](image_placeholder)

# Notations

Likelihood Ratio (LR) that compares two fully-specified (discrete) models is simply the ratio of the likelihood for the two models given data $x$: 

$$
LR(M_1,M_0):=\frac{L(M_1)}{L(M_0)} = \frac{p(x|M_1)}{p(x|M_0)}
$$

Large values of $LR(M_1,M_0)$ indicate that the data are much more probable under model $M_1$ than under model $M_0$, and so indicate support for $M_1$. Conversely, small values of LR indicate support for model $M_0$.

Note that for fully specified models, the likelihood ratio is also known as the **"Bayes Factor" (BF)**. In this case, they are just two different names for the same thing.



<!-- 

The Likelihood Ratio Test (LRT) is a statistical test used to compare the fit of two models: a **null model** and an **alternative model**. The null model usually represents a simpler hypothesis (e.g., no effect), and the alternative model represents a more complex hypothesis (e.g., an effect exists). 

The basic idea is to test whether the data provides enough evidence to reject the null hypothesis in favor of the alternative hypothesis. Note that the Likelihood Ratio Test (LRT) is primarily a frequentist approach to hypothesis testing, not Bayesian.


The Likelihood Ratio Test statistic is defined as:

$$
\Lambda = -2 \log \left( \frac{L(\text{null})}{L(\text{alternative})} \right)
$$

Where:
- $L(\text{null})$ is the likelihood of the data under the null model,
- $L(\text{alternative})$ is the likelihood of the data under the alternative model.

The test statistic $\Lambda$ is compared to a **chi-squared distribution** with degrees of freedom equal to the difference in the number of parameters between the two models.

### Bayes Factor (BF)
In the Bayesian framework, we approach model comparison slightly differently than the frequentist approach. Instead of using the likelihood ratio to compute a p-value for hypothesis testing, Bayesian methods typically use the **Bayes Factor** (BF) to compare the models.

The **Bayes Factor** compares the marginal likelihoods (or evidence) of the models:

$$
\text{BF} = \frac{P(\text{data} | \text{model 1})}{P(\text{data} | \text{model 2})}
$$

Where:
- $P(\text{data} | \text{model 1})$ and $P(\text{data} | \text{model 2})$ are the marginal likelihoods (or evidence) of the models. In the frequentist LRT, this would correspond to the likelihoods of the two models.

In this way, the Bayes Factor provides a measure of the relative evidence for one model over another, rather than using a p-value to decide whether to reject a model.

 -->

# Example

## Example 1 -- allele frequency

Here we use the same example 1 in the `6_1_likelihood.ipynb` notebook, and we have calculated the likelihood for the two models $M_S$ and $M_F$.

### Calculate LR

In [184]:
x = c(1,0,1,0,0,1)

fS = c(0.40, 0.12,0.21,0.12,0.02,0.32)
fF = c(0.8,0.2,0.11,0.17,0.23,0.25)

L = function(f,x){ prod(f^x*(1-f)^(1-x)) }

In [185]:
L(fS,x)
L(fF,x)

[1] 0.02039955

[1] 0.01124816

The **likelihood ratio (LR)** is simply the ratio of those two values:

In [186]:
L(fS,x)/L(fF,x)

[1] 1.81359

### How to interpret LR?

> So $LR(M_S,M_F;x)$ is 1.8135904. This means that the data favor the tusk coming from a savanna elephant by a factor of about 1.8. This is a fairly modest factor – not large enough to draw a convincing conclusion. We will have more to say about interpreting LRs, and what values might be considered “convincing” later.
>
> Note that we have deliberately focused on the likelihood ratio, and not the actual likelihood values themselves. This is because actual likelihood values are generally not useful - it is only the ratios that matter when comparing the models. One way of thinking about this is that the actual likelihood values are very context dependent, and so likelihoods from different data sets are not comparable with one another. However, the meaning of the likelihood ratio is in some sense consistent across contexts: LR =1.8 means that the data favour the first model by a factor of 1.8 whatever the context.

## Example 2 -- concertration of protein in blood

Here we use the same example 2 in the `6_1_likelihood.ipynb` notebook, and we have calculated the likelihood for the two models $M_0$ (normal individual group) and $M_1$ (diseased individual group).

### Calculate LR

In [187]:
X_val=4.02
# dgamma(x, shape, rate = 1, scale = 1/rate, log = FALSE) returns the Density for the Gamma distribution with parameters shape and scale at x
y0_val = dgamma(X_val,scale=0.5,shape=2)
y1_val = dgamma(X_val,scale=1,shape=2)
y0_val
y1_val

[1] 0.005182728

[1] 0.07217092

In [188]:
y1_val/y0_val

[1] 13.92528

### Interpretation

The $LR(M_1,M_0;x)$ is 13.9, i.e., the data favours the individual being diseased ($M_1$) over being normal ($M_0$) by a factor of approximately 14.



## Summary
- The **"likelihood ratio"** that compares two fully-specified (discrete) models is simply the ratio of the likelihood for the two models given data $x$: 

$$
LR(M_1,M_0):=\frac{L(M_1)}{L(M_0)} = \frac{p(x|M_1)}{p(x|M_0)}
$$

where $L(M)$ denotes the likelihood for model $M$ under data $x$.

- We noticed that in the first example, LR is 1.8 and in the second example, LR is 14. However, as Matthew Stephens pointed out:
> It is crucial to recognize that the answer to this question has to be context dependent. In particular, the extent to which we should be “convinced” by a particular LR value has to depend on the relative plausibility of the models we are comparing. For example, in statistics there are many situations where we want to compare models that are not equally plausible. Suppose that model $M_1$ is much less plausible than $M_0$. Then we must surely demand stronger evidence from the data (larger LR) to be “convinced” that it arose from model $M_1$ rather than $M_0$, than in contexts where $M_1$ and $M_0$ are equally plausible.


# Recommended Reading

- Section *Likelihood Ratio and Likehood* in [FiveMinuteStats](http://stephens999.github.io/fiveMinuteStats/index.html) by Matthew Stephens
